### DISCUSSION TITLE
Investigation into Idiosyncracy in Training/Validation Loss Relationship

### DISCUSSION THREAD

**Discussion by MarioShontz** [emelex-ai/BRIDGE#103] [General]
## The problem
A valid model and accuracy scoring method will generally produce a consistent pattern in training and validation scores - Typically, the validation accuracy will be  less than (but theoretically could be equal to) the training accuracy, so long as the accuracies are calculated soundly and the dataset randomly sampled. 

Contrary to this principle, our phon_word_accuracy on some (but not all) training runs have higher accuracy on the validation set than the training set.
![image](https://github.com/emelex-ai/ConnTextUL/assets/9813567/0acb0eaf-c6b2-43c5-885c-dc343d93a1b8)

In fact, for this same run, the direct loss values follow the same idiosyncratic pattern
![image](https://github.com/emelex-ai/ConnTextUL/assets/9813567/9adab625-9caf-46c4-af39-b17b42e12f79)

This doesn't make quite sense, and so I am in search of a possible solution.

## Looking for a solution
I will be looking at the calculation of this loss metric, and particularly how it depends on certain hyperparameters. This issue arose only when we trained with the recent set of hyperparameters, so this will be where I look.

**Comment by mathnathan** [comment]
This is great. Thanks for the quick write-up

I would also suggest examining all metrics side-by-side. For example, the loss function that the gradients are calculated from operate at a per phoneme-segment level. In other words, it is the average error incurred in predicting whether each of the 33 phoneme segments are active or not in one phonology vector. There are two more metrics that build on this providing different levels of introspection. Next we have the phonology vector accuracy, which is, the average number of phonology vectors that are correct (meaning all phoneme segments are correct) in all words across a batch. The highest level is the phonology word accuracy, which is, the average number of words that are correct (meaning all phonology vectors are correct) within an entire batch. 

Looking at all of these side-by-side and incorporating the batch size and the max phonology length in a dataset could provide additional insights.

**Reply by MarioShontz** [reply]
Thank you, that is useful to know. I'll start there.
